src/libopenrave/robotconnectedbody.cpp

```diff
@@ -711,8 +711,27 @@ void RobotBase::_ComputeConnectedBodiesInformation()
             for( std::string& forcedAdjacentLink : plink->_info._vForcedAdjacentLinks ) {
                 forcedAdjacentLink = connectedBody._nameprefix + forcedAdjacentLink;
             }
+
             _InitAndAddLink(plink);
             connectedBody._vResolvedLinkNames[ilink].first = plink->_info._name;
+
+            // Set all links from connected bodies with -1 to be adjacent, otherwise, we can detect self collision when in indeterminable state.
+            if (connectedBody.IsActive() == -1) {
+                vector<ConnectedBodyPtr>::iterator itPrevConnectedBody = itconnectedBody;
+                while(itPrevConnectedBody != _vecConnectedBodies.begin()) {
+                    --itPrevConnectedBody;
+                    ConnectedBody& prevConnectedBody = **itPrevConnectedBody;
```

Comment:
Shouldn't it be applied only on pairs of connected bodies which share a same attaching link?

Reply:
It does seem safer just in case we have multiple tool changers, all in indeterminable state, that could physically collide together from the robot's motion.